src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -434,12 +435,13 @@ CoreWorkerDirectTaskSubmitter::GetOrConnectLeaseClient(
 void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
     const SchedulingKey &scheduling_key, const rpc::Address *raylet_address) {
   auto &scheduling_key_entry = scheduling_key_entries_[scheduling_key];
-  auto &pending_lease_request = scheduling_key_entry.pending_lease_request;
 
-  if (pending_lease_request.first) {
-    // There's already an outstanding lease request for this type of task.
+  if (scheduling_key_entry.pending_lease_request_task_to_raylet.size() ==
```

Comment:
```suggestion
  if (scheduling_key_entry.pending_lease_request_task_to_raylet.size() >=
```

Prefer defensive code

Reply:
I actually added a RAY_CHECK in the next line to make sure we never exceeds the limit.